src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs

```diff
@@ -98,6 +98,8 @@ internal static void ExtractRelativeToDirectory(this ZipArchiveEntry source, str
             // Note that this will give us a good DirectoryInfo even if destinationDirectoryName exists:
             DirectoryInfo di = Directory.CreateDirectory(destinationDirectoryName);
             string destinationDirectoryFullPath = di.FullName;
+            if (!destinationDirectoryFullPath.EndsWith(Path.DirectorySeparatorChar))
```

Comment:
Does this also need to deal with alternate separators?

Reply:
No, the paths are all normalized. In the test I'm actually using the "alternate" for Windows and it failed/passed as expected.